### PR TITLE
use typescript

### LINKED ISSUE #36: use TypeScript

### PR DESCRIPTION
**Pull request by shoota**
Closes #36

### CODE REVIEW

**Comment by 9renpoto** [comment]
Please migrate test to TypeScript files

File: webpack.config.js [hunk 1]
```diff
@@ -3,15 +3,19 @@ const path = require('path')
...
     path: path.join(__dirname, 'example/'),
     filename: '[name].bundle.js'
   },
+  resolve: {
+    extensions: ['.ts', '.tsx', '.js']
+  },
   module: {
     loaders: [
-      { test: /\.js$/, loader: 'babel-loader', exclude: /node_modules/ }
+      { test: /\.js$/, loader: 'babel-loader', exclude: /node_modules/ },
```
Comment (9renpoto) [inline webpack.config.js]: is this need ?

File: src/index.ts [hunk 2]
```diff
@@ -25,12 +18,12 @@ function findOrCreate (
   const domainParts = removeNaked().split('.')
   const subDomain: any = domainParts[domainParts.length - 1]
   // eslint-disable-next-line eqeqeq
-  if (domainParts.length === 4 && parseInt(subDomain, 10) == subDomain) {
+  if (domainParts.length === 4 && parseInt(subDomain, 10) === subDomain) {
```
Comment (9renpoto) [inline src/index.ts]: revert this. because  `typeof subDomain === 'string'`

File: webpack.config.js (hunk 1 above)
Comment (shoota) [reply webpack.config.js]: it is unnecessary. I will delete line.

File: wdio.conf.js [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
+  specs: ['./test/*.ts'],
   capabilities: [
     {
       browserName: 'firefox'
     }
   ],
   exclude: [],
   bail: 0,
-  sync: true,
+  sync: false,
```
Comment (shoota) [inline wdio.conf.js]: http://webdriver.io/guide/testrunner/configurationfile.html

> If you still want to run your tests in an async way using promises you can set the sync command to false.

File: package.json [hunk 4]
```diff
@@ -31,6 +35,9 @@
     "prettier-eslint-cli": "^4.1.1",
     "rimraf": "^2.6.1",
     "serve-static": "^1.12.3",
+    "ts-loader": "^2.3.7",
+    "ts-node": "^3.3.0",
```
Comment (shoota) [inline package.json]: for execute static-server, and require wdio compiler

File: src/index.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+function removeNaked (): string {
+  const domain = `${location.hostname}`
+  return domain.indexOf('www.') === 0 ? domain.substring(4) : domain
+}
+
+function setCookie (
+  domainParts: Array<string>,
+  name: string,
+  options: cookies.CookieAttributes,
+  data: string
```
Comment (9renpoto) [inline src/index.ts]: only `string` ?

File: test/helpers/server.ts [hunk 6]
```diff
@@ -1,10 +1,12 @@
-/* @flow */
+declare function require (path: string): any
+declare var __dirname
+
 const http = require('http')
```
Comment (9renpoto) [inline test/helpers/server.ts]: `import * as http from 'http'` ?

File: test/helpers/server.ts [hunk 7]
```diff
@@ -1,10 +1,12 @@
-/* @flow */
+declare function require (path: string): any
```
Comment (9renpoto) [inline test/helpers/server.ts]: this declare is not necessary?

File: test/index.test.ts [hunk 8]
```diff
@@ -1,14 +1,13 @@
-/* @flow */
-import assert from 'assert'
-import { describe, before, beforeEach, it } from 'mocha'
+import 'mocha'
+import * as assert from 'power-assert'
```
Comment (9renpoto) [inline test/index.test.ts]: `power-assert` alias `assert`

File: src/index.ts (hunk 5 above)
Comment (shoota) [reply src/index.ts]: Yes. To set cookie,  value is required.

Comment (9renpoto) [reply src/index.ts]: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/js-cookie/index.d.ts#L44

value maybe `string | object`

File: test/helpers/server.ts (hunk 7 above)
Comment (shoota) [reply test/helpers/server.ts]: change to what using `@types/node` to resolve it.

File: src/index.ts (hunk 5 above)
Comment (shoota) [reply src/index.ts]: oh. now, I checked js-cookie interface and will extend interface for fix this.

Comment (9renpoto) [reply src/index.ts]: I am waiting for a new PR :)

**Pull request merged**